app/models/tweet.rb

```diff
@@ -59,10 +77,33 @@ def tweetor_name
   end
 
   def tweetor_link
-    "https://twitter.com/#{tweetor}"
+    twitter_account&.twitter_account_url || "https://twitter.com/#{tweetor}"
   end
 
   def tweet_link
-    "https://twitter.com/#{tweetor}/status/#{twitter_id}"
+    if twitter_account.present?
+      [twitter_account.twitter_account_url, "status", twitter_id].join("/")
+    else
+      "https://twitter.com/#{tweetor}/status/#{twitter_id}"
+    end
+  end
+
+  def details_hash
+    @details_hash ||= begin
```

Comment:
Is begin idiomatic here? I feel like I've only seen begin for rescuing, and it seems surprising.

Reply:
> Is begin idiomatic here?


'tis

https://www.rubydoc.info/gems/rubocop/RuboCop/Cop/Style/MultilineMemoization
https://medium.com/helpful-tips/memoization-patterns-in-ruby-9efacbcee130
https://www.justinweiss.com/articles/4-simple-memoization-patterns-in-ruby-and-one-gem/
http://www.biggillystyle.com/2014/11/28/ruby-memoization-techniques.html
http://vaidehijoshi.github.io/blog/2015/11/10/methods-to-remember-things-by-ruby-memoization/

Reply:
Cool!